src/libraries/Common/src/System/IO/PathInternal.CaseSensitivity.cs

```diff
@@ -31,13 +31,30 @@ internal static StringComparison StringComparison
         /// <summary>
         /// Determines whether the file system is case sensitive.
         /// </summary>
+        private static bool GetIsCaseSensitive()
+        {
+#if NET6_0_OR_GREATER
+            // Return a constant for mobile platforms without resorting to I/O
+            if (OperatingSystem.IsMacOS() || OperatingSystem.IsMacCatalyst() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS())
+                return true;
+            if (OperatingSystem.IsBrowser())
+                return false;
+            if (OperatingSystem.IsAndroid())
+                return false;
+#endif
+            return GetIsCaseSensitiveByProbing();
```

Comment:
```suggestion
#if MS_IO_REDIST
            return false; // Windows is always case-insensitive
#else
    return OperatingSystem.IsWindows() || OperatingSystem.IsMacOS() || OperatingSystem.IsMacCatalyst() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS();
#endif
```

Reply:
If no probing is needed could we remove the implementation from Path and call into PathInternal all the time?

Reply:
Yes, that is a good idea. We can even delete the property on Path and change the few places that use it to call PathInternal instead.